tfx/components/example_gen/utils.py

```diff
@@ -300,19 +332,34 @@ def _retrieve_latest_span_version(
             'Cannot find %s number from %s based on %s' %
             (VERSION_PROPERTY_NAME, file_path, split_regex_pattern))
 
-    if latest_span is None or span_int > int(latest_span):
+    if (latest_span_elems is None
+        or span_elems_int > [int(elem) for elem in latest_span_elems]):
```

Comment:
compare two list seems not ideal,  convert the date to span_int and still use span_int for comparison?

Reply:
fixed. now does conversion for each file path